pkgs/by-name/re/regina-normal/package.nix

```
@@ -0,0 +1,99 @@
+{
+  lib,
+  fetchurl,
+  python3,
+  cmake,
+  cppunit,
+  doxygen,
+  gmp,
+  graphviz,
+  jansson,
+  libxml2,
+  libxslt,
+  perl,
+  pkg-config,
+  popt,
+  qt6,
+  shared-mime-info,
+  stdenv,
+  lmdb,
+  databaseLibrary ? lmdb, # can be built with tokyocabinet instead
+  highDim ? false, # whether or not to include triangulations of dimension 9-15
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "regina-normal";
+  version = "7.3.1";
+
+  # Signed source tarball
+  src = fetchurl {
+    url =
+      with finalAttrs;
+      "https://github.com/regina-normal/regina/releases/download/regina-${version}/regina-${version}.tar.gz";
+    hash = "sha256-lfUJcNrK13DtwNOvlerwbEndalRzJ/WKtVQBx9Qcg/o=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    cppunit
+    doxygen
+    libxslt
+    pkg-config
+  ];
+
+  buildInputs = [
+    databaseLibrary
+    gmp
+    graphviz
+    jansson
+    libxml2
+    perl
+    popt
+    python3
+    qt6.qtbase
+    qt6.qtsvg
+    qt6.wrapQtAppsHook
+    shared-mime-info
+  ];
+
+  prePatch = ''
+    substituteInPlace python/regina/CMakeLists.txt \
+      --replace-fail '$ENV{DESTDIR}''${Python_SITELIB}' \
+      "${placeholder "out"}/${python3.sitePackages}"
+    substituteInPlace python/regina-python.in \
+      --replace-fail "my \$python_lib_dir = '''" \
+      "my \$python_lib_dir = '${placeholder "out"}/${python3.sitePackages}'"
+    patchShebangs utils/testsuite/*.{in,test,sh} python/testsuite/test*.in python/regina-python.in
+  '';
+
+  env = {
+    PERL_PYLIBDIR = "${placeholder "out"}/${python3.sitePackages}";
+    BASH_CMAKE_SOURCE_DIR = "${finalAttrs.src}";
+  };
+
+  cmakeFlags = [
+    "-DREGINA_INSTALL_TYPE=XDG"
+    "-DPython_EXECUTABLE=${python3.interpreter}"
+    "-DPython_SITELIB=${python3.sitePackages}"
+    "-DPYLIBDIR=${placeholder "out"}/${python3.sitePackages}"
+    "-DCMAKE_POLICY_DEFAULT_CMP0153=OLD"
+  ]
+  ++ lib.optionals highDim [ "-DHIGHDIM=1" ];
+
+  doInstallCheck = true;
+
+  installCheckPhase = ''
+    runHook preInstallCheck
+    make test
+    runHook postInstallCheck
+  '';
```

```suggestion
  installCheckTarget = "test";
```
I think specifying `installCheckTarget` does exactly this.